src/CurrentRoute.php

```diff
@@ -75,6 +80,25 @@ public function setUri(UriInterface $uri): void
             $this->uri = $uri;
             return;
         }
-        throw new RuntimeException('Can not set URI since it was already set.');
+        throw new LogicException('Can not set URI since it was already set.');
+    }
+
+    public function getParameters(): array
+    {
+        return $this->parameters;
+    }
+
+    public function getParameter(string $name, $default = null): ?string
+    {
+        return $this->parameters[$name] ?? $default;
+    }
+
+    public function setParameters(array $parameters): void
+    {
+        if ($this->parameters === []) {
+            $this->parameters = $parameters;
+            return;
+        }
+        throw new LogicException('Can not set parameters since it was already set.');
     }
```

Comment:
```suggestion
    public function setArguments(array $arguments): void
    {
        if ($this->arguments === []) {
            $this->arguments = $arguments;
            return;
        }
        throw new LogicException('Can not set arguments since they were already set.');
    }
```